src/main/java/com/eerussianguy/blazemap/feature/maps/AtlasExporter.java

```diff
@@ -0,0 +1,144 @@
+package com.eerussianguy.blazemap.feature.maps;
+
+import com.eerussianguy.blazemap.api.BlazeRegistry;
+import com.eerussianguy.blazemap.api.maps.Layer;
+import com.eerussianguy.blazemap.api.maps.MapType;
+import com.eerussianguy.blazemap.api.maps.TileResolution;
+import com.eerussianguy.blazemap.api.util.RegionPos;
+import com.eerussianguy.blazemap.engine.BlazeMapAsync;
+import com.eerussianguy.blazemap.engine.StorageAccess;
+import com.eerussianguy.blazemap.engine.client.BlazeMapClientEngine;
+import com.eerussianguy.blazemap.engine.client.LayerRegionTile;
+import com.eerussianguy.blazemap.util.Colors;
+import com.mojang.blaze3d.platform.NativeImage;
+import net.minecraft.client.Minecraft;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.Level;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AtlasExporter {
+    private static Task task = null;
+
+    public static synchronized void export(Task task) {
+        if(AtlasExporter.task == null) {
+            AtlasExporter.task = task;
+            BlazeMapAsync.instance().clientChain.runOnDataThread(AtlasExporter::export);
+        }
```

Comment:
There should probably be an `else` condition here for if a second export is started while the first is ongoing. Eg, starting an export of the Aerial Map, swap to Terrain Height MapType, try export from Terrain Height while Aerial Map is still exporting, get confused why it didn't work.

This can be done in a follow-up PR.

Reply:
The reasoning for this is that since Export is heavy and is going to monopolize one thread, you can only have one at a time. And since the only place you can trigger it from has feedback for the current one, it felt unnecessary. I can probably make the current export flash in red when F12 is pressed though, for further indication.

Later.

Reply:
Flashing red added in working branch